File: services/feed/internal/lua/revrange.go
package interlua

type revRange struct {
	name     string
	function string
}

func (r *revRange) Name() string {
	return r.name
}
func (r *revRange) Function() string { return r.function }

var revRangeScript *revRange

func init() {
	revRangeScript = &revRange{}
	revRangeScript.name = "revRange"
	revRangeScript.function = `
local key=KEYS[1]
local limit=ARGV[1]

local exists=redis.call("EXISTS",key)
if exists==0
then return nil
end

local res=redis.call("ZREVRANGE",0,tonumber(limit),"WITHSCORES")
return res
`
}

func GetRevRangeScript() *revRange {
	return revRangeScript
}

File: services/relation/internal/script/revrange.lua
redis =require"redis"
local KEYS={}
local ARGV={}
--上述三行为了防止报错，在使用时将其删除
local key=KEYS[1]
local all=ARGV[1]
local b=ARGV[2]
local e=ARGV[3]

local exists=redis.call("EXISTS",key)
if exists==0
    then return nil
end

if all=="true"
    then
    local res=redis.call("ZREVRANGE",key,0,-1)
    return res
end

local res=redis.call("ZREVRANGE",key,b,e)
return res